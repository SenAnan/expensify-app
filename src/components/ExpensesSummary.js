import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = props => {
	const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
	return (
		<div>
			<h2>
				Viewing {props.expenses.length} {expenseWord} totalling{' '}
				{numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}
			</h2>
		</div>
	);
};

const mapStateToProps = state => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
