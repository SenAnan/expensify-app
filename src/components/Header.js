import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

const Header = (props) => (
	<header>
		<h1>Expensify</h1>
		<ul>
			<li>
				<NavLink to="/dashboard" activeClassName="is-active">
					Dashboard
				</NavLink>
			</li>
			<li>
				<NavLink to="/create" activeClassName="is-active">
					Add Expense
				</NavLink>
			</li>
			<li>
				<NavLink to="/help" activeClassName="is-active">
					Help
				</NavLink>
			</li>
			<button onClick={props.startLogout}>Logout</button>
		</ul>
	</header>
);

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
