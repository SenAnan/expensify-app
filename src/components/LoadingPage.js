import React from 'react';

const LoadingPage = (props) => {
	return (
		<div className="loader">
			<img src="/images/loader.gif" alt="Loader" className="loader__image" />;
		</div>
	);
};

export default LoadingPage;
