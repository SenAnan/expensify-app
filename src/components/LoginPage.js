import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends Component {
	render() {
		return (
			<div className="LoginPage">
				<div className="LoginPage__container">
					<h1 className="LoginPage__title">Expensify</h1>
					<p className="LoginPage__text">It's time to get your expenses under control</p>
					<button className="LoginPage__button button" onClick={this.props.startLogin}>
						Login with Google
					</button>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
