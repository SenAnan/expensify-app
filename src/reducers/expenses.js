const expensesDefaultState = [];

const expensesReducer = (state = expensesDefaultState, action) => {
	switch (action.type) {
		case 'ADD_EXPENSE':
			return [...state, action.expense];
		case 'REMOVE_EXPENSE':
			return state.filter((el) => el.id != action.id);
		case 'EDIT_EXPENSE':
			return state.map((el) => {
				if (el.id == action.id) {
					return {
						...el,
						...action.updates,
					};
				} else {
					return el;
				}
			});
		case 'SET_EXPENSES':
			return action.expenses;
		default:
			return state;
	}
};

export default expensesReducer;
