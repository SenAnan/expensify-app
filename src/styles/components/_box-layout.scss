.LoginPage {
	background-image: url('/images/bg.jpg');
	background-size: cover;
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100vh;
	width: 100vw;

	&__container {
		background-color: fade-out(white, 0.15);
		border-radius: 3px;
		justify-content: center;
		align-items: center;
		text-align: center;
		padding: $l-size $m-size;
	}

	&__button {
		width: 80%;
		height: 50px;
		background-color: $blue;
		border: none;
		color: white;
		font-size: $font-size-large;
		padding: $s-size;
		font-weight: 300;
	}

	&__text {
		line-height: 1;
		padding-bottom: $m-size;
	}
}
