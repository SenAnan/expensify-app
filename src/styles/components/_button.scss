.button {
	height: 50px;
	background-color: $blue;
	border: none;
	color: white;
	font-size: $font-size-large;
	padding: $s-size;
	font-weight: 300;

	&--secondary {
		background-color: #888;
	}
}
