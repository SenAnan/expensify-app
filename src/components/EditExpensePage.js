import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
	onSubmit = (expense) => {
		this.props.onSubmit(expense);
		this.props.history.push('/');
	};
	onRemove = () => {
		this.props.startRemoveExpense();
		this.props.history.push('/');
	};

	render() {
		return (
			<div>
				<ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}></ExpenseForm>
				<button onClick={this.onRemove}>Remove</button>
			</div>
		);
	}
}

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find((el) => el.id == props.match.params.id),
	};
};

const mapDispatchToProps = (dispatch, props) => {
	return {
		onSubmit: (expense) => dispatch(startEditExpense(props.match.params.id, expense)),
		startRemoveExpense: () => dispatch(startRemoveExpense({ id: props.match.params.id })),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
