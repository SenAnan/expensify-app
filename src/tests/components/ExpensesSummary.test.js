import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should correctly render 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
	expect(wrapper).toMatchSnapshot();
});

test('Should correctly render multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
	expect(wrapper).toMatchSnapshot();
});
