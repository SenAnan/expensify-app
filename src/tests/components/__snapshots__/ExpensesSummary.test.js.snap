// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling
     
    $1,095.00
  </h2>
</div>
`;

exports[`Should correctly render multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling
     
    $1,141.95
  </h2>
</div>
`;
